src/utils/googleTagManager.ts

```diff
@@ -155,14 +165,61 @@ export const trackEvent = ({
   action: string
   label?: EventLabel
 }): void => {
-  const dataLayer = {
+  const dataLayer: EventDataLayer = {
     event,
     chainId: _getChainId(),
     eventCategory: category,
     eventAction: action,
     eventLabel: tryParse(label),
   }
 
+  track(dataLayer)
+}
+
+const ETHEREUM_ADDRESS_REGEX = /0x[a-fA-F0-9]{40}/g
+const ETHEREUM_ADDRESS = 'ETHEREUM_ADDRESS'
+
+type SafeAppEventDataLayer = {
+  event: GTM_EVENT
+  chainId: string
+  safeAppName: string
+  safeAppMethod: string
+  safeAppParams?: string
+  safeAppEthMethod?: string
+  safeAppDeprecatedMethod: boolean
+  safeAppSDKVersion?: string
+}
+
+export const trackSafeAppEvent = ({
+  event,
+  name,
+  method,
+  params,
+  sdkVersion,
+  deprecated = false,
+}: {
+  event: GTM_EVENT
+  name: string
+  method: string
+  params: any
+  sdkVersion?: string
+  deprecated?: boolean
+}): void => {
+  const dataLayer: SafeAppEventDataLayer = {
+    event,
+    chainId: _getChainId(),
+    safeAppName: name,
+    safeAppMethod: method,
+    safeAppParams: params ? JSON.stringify(params).replaceAll(ETHEREUM_ADDRESS_REGEX, ETHEREUM_ADDRESS) : undefined,
+    safeAppEthMethod: params?.call || undefined,
+    safeAppDeprecatedMethod: deprecated || Object.values<string>(LegacyMethods).includes(method),
```

Comment:
I believe google filters are powerful, and we could calculate deprecated methods in the Google UI. Do we really need to send this?

Reply:
You mean adding the methods in the report when needed? 

For me is ok if you prefer that. This came from a conversation with dani yesterday 

@dasanra what do you think?